File: multiples_in_range/python/lib/number_multiples/number_multiples_test.py
import unittest
from lib.number_multiples.number_multiples import NumberMultiples


class NumberMultiplesTest(unittest.TestCase):
    """Test class NumberMultiples"""
    def setUp(self):
        """
        This method is called before each test
        """
        self.multiples = NumberMultiples(n=10)

    def tearDown(self):
        """
        This method is called after each test
        """
        pass

    def test_count(self):
        self.assertRaises(TypeError, lambda: self.multiples.count_multiples_of('a'))
        self.assertEqual(self.multiples.count_multiples_of(3), 3)
        self.assertEqual(self.multiples.count_multiples_of('3'), 3)

    def test_list(self):
        expected = {
            1: 3,
            2: 6,
            3: 9
        }
        self.assertRaises(TypeError, lambda: self.multiples.list_multiples_of('a'))
        self.assertEqual(self.multiples.list_multiples_of(3), expected)
        self.assertEqual(self.multiples.list_multiples_of('3'), expected)

    def test_sum(self):
        self.assertRaises(TypeError, lambda: self.multiples.sum_multiples_of('a'))
        self.assertEqual(self.multiples.sum_multiples_of(3), 18)
        self.assertEqual(self.multiples.sum_multiples_of('3'), 18)
        self.assertRaises(TypeError, lambda: self.multiples.sum_multiples_of('a', 5))
        self.assertEqual(self.multiples.sum_multiples_of('3', 5), 23)
        self.assertEqual(self.multiples.sum_multiples_of('3', '5'), 23)

File: multiples_in_range/ruby/lib/number_multiples_test.rb
require_relative 'number_multiples'
require 'test/unit'

# Testing number multiples
class NumberMultiplesTest < Test::Unit::TestCase
  def setup
    @sum = NumberMultiples.new(10)
  end

  def test_count
    assert_equal(TypeError, @sum.count_multiples_of('a'))
    assert_equal(3, @sum.count_multiples_of(3))
    assert_equal(3, @sum.count_multiples_of('3'))
  end

  def test_list
    expected = {
      1 => 3,
      2 => 6,
      3 => 9
    }
    assert_equal(TypeError, @sum.list_multiples_of('a'))
    assert_equal(expected, @sum.list_multiples_of(3))
    assert_equal(expected, @sum.list_multiples_of('3'))
  end

  def test__sum
    assert_equal(TypeError, @sum.sum_multiples_of(''))
    assert_equal(18, @sum.sum_multiples_of(3))
    assert_equal(18, @sum.sum_multiples_of('3'))
    assert_equal(TypeError, @sum.sum_multiples_of('a', 5))
    assert_equal(23, @sum.sum_multiples_of(3, 5))
    assert_equal(23, @sum.sum_multiples_of('3', 5))
  end

end